### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by addod19**

### CODE REVIEW

File: enum.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+    return as_array unless as_array.is_a?(Array)
+
+    return to_enum(:my_inject) unless block_given?
+
+    result = !args.empty? ? args[0] : as_array[0]
+
+    1.upto(len - 1) { |i| result = yield(result, as_array[i]) }
+    result
+  end
```
Comment (maelfosso) [inline enum.rb:142]: Sorry :disappointed: there is a bug
```ruby
irb(main):043:0> [5,6,7,8,9,10].inject(5) { |sum, n| sum + n } 
=> 50
irb(main):044:0> [5,6,7,8,9,10].my_inject(5) { |sum, n| sum + n } 
=> 45
irb(main):045:0> [5,6,7,8,9,10].inject(5) { |prod, n| prod * n } 
=> 756000
irb(main):046:0> [5,6,7,8,9,10].my_inject(5) { |prod, n| prod * n } 
=> 151200
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @addod19 ,

# GReat work !! :smile:  You deserved a standing ovation :clap: 

There is just one bug to fix in the `inject` function. Take a lot at the comment below please.

Kindly fix it a submit a new code review request.
Thanks

File: enum.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      0.upto(len - 1) { |i| result << proc[0].call(as_array[i]) }
+      result
+    end
+    return to_enum(:my_map) unless block_given?
+
+    0.upto(len - 1) { |i| result << yield(as_array[i]) }
+    result
+  end
+
+  def my_inject(*args)
```
Comment (bellom) [inline enum.rb:134]: ## Opps, we still need to work on `inject`, got the below error;
image
Note; you can always user `my_each` method in the file  to make your work easier.

**Review by bellom — COMMENTED** [review]
...

**Review by bellom — CHANGES_REQUESTED** [review]
Hi @addod19,
**NOTE: Always read the pathwright instructions before starting the project to avoid the need of sending another code review, according to the pathwright you are to use Git Flow not GitHub Flow.**
_Short Explanation Hint:
GitHub Flow: Pull Requests is raise from feature branches to the master.
Git Flow: Pull Requests is raise from feature branches to the development._

**Make sure to use Git flow in the next project and also add Linter's check on your PR page.**

_Just one thing to do, checkout my comment._

**Kindly re-submit for review after the `my_inject` method :slightly_smiling_face: Thumps -up await you....**

**Review by bellom — CHANGES_REQUESTED** [review]
Hi @addod19, 
## Wonderful work using `Git flow` and Good job correcting all the issues from previous reviews.
**Just one thing to do which is `_Required_`;**
1.  Fix all errors on linter, checkout this [linters-config#troubleshooting](https://github.com/microverseinc/linters-config#troubleshooting) for help. Example below;
image

**_Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you_**

**Review by bolah2009 — APPROVED** [review]
## It's a :+1: from me @addod19  :smile: 

Nice Job implementing all requested changes

#### I strongly recommend updating this PR description.
   update-PR-description

## Go ahead and merge to master :rocket: 

#### Also, don't forget to submit project completion form. :blush: 

Good luck with your next project. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Comment by addod19** [comment]
# Custom Enumerable Methods
This project is setup with .rubocop.yml and .stickler.yml file. The code is reviewed

**Pull request merged**